### DISCUSSION TITLE
Website using ln-crawler

### DISCUSSION THREAD

**Discussion by jere344** [lncrawl/lightnovel-crawler#1408] [Ideas]
Hello, I started making a website with flask that allow users to directly add novels using lightnovel-crawler.
I used lightnovelpub design (minus ads) which took it from lightnovelworld, so I don't know who made in first.
I don't know if you are interested in merging since I stole the design.

It is simply a bot and can be run with ```python lncrawl --bot web```
You can check it on the [flask branch of my fork from lncrawler](https://github.com/jere344/lightnovel-crawler/tree/flask) 

It use flask and waitress and Flask-Minify as dependencies.

The architecture for light novels is reversed, it is ```Lightnovels/novel/source``` as I don't want the same novel to appear twice. You choose the novel first and then you can switch source.
It still need some work but is functional right now.

Also to make the website better I would like to add the possibility to scrap summary in sources, (and perhaps also tags and language).

What do you think?

**Comment by dipu-bd** [comment]
Nice! I have long been planning to add a webserver as a bot to this app. But I was getting pushed back for the lack of time and energy.

I will check your changes and merge it with the dev branch if all is good.

**Comment by jere344** [comment]
Glad you are interested !

You can make any change or ask me to change anything, I have no experience in production so it probably is a mess.


The flask website is usable right now (still lacks some features like allowing users to rate novels and sources) but I dropped it and switched to react. It mean that I must remake everything but css as I can't just use lightnovelworld scripts like before.
I am working on it [on the react branch of my fork](https://github.com/jere344/lightnovel-crawler/tree/react)

**Comment by jere344** [comment]
It still has a few known bug, stability issues and is missing https but I am hosting the website on http://lncrawler.monster/

PS : if you go on http**s**://lncrawler.monster/ you will go on a different website, I will fix that later

**Comment by jere344** [comment]
Are you interested in adding summary, language and tags as additional optional metadata ? I added it on my fork for the website, but supporting it myself for every sources and dealing with conflicts after every update would be too hard.
The language can use the source folder by default and only need to be manually set for sources with multiple languages (in multi folder).

I can make a clean fork with only theses changes and open a pull request if you are interested.

**Reply by dipu-bd** [reply]
I am not sure how you want to implement this feature. If you have already done it, please make a pull request. 

Otherwise we can open an issue discussing about how to add these extra fields and migrating the existing sources.
